feat/src/main/java/com/example/feat/course_detail/components/CourseDetailToolbar.kt

```diff
@@ -0,0 +1,116 @@
+package com.example.feat.course_detail.components
+
+import androidx.compose.animation.core.animateDpAsState
+import androidx.compose.animation.core.animateFloatAsState
+import androidx.compose.animation.core.tween
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.lazy.LazyListState
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowBack
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.layout.layoutId
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.constraintlayout.compose.ExperimentalMotionApi
+import androidx.constraintlayout.compose.MotionLayout
+import androidx.constraintlayout.compose.MotionScene
+import coil.compose.AsyncImage
+import com.example.feat.R
+import com.example.core.lib.constants.DesignSystem
+
+@OptIn(ExperimentalMotionApi::class)
+@Composable
+fun CourseDetailToolbar(
+    lazyScrollState: LazyListState,
+    backgroundImageUrl: String? = null,
+    title: String? = null
+) {
+    val context = LocalContext.current
+    val configuration = LocalConfiguration.current
+    val screenHeight = configuration.screenHeightDp
+
+    val motionScene = remember {
+        context.resources.openRawResource(R.raw.collapse_toolbar).readBytes().decodeToString()
```

Comment:
Using `readBytes().decodeToString()` can potentially lead to issues with large files and memory usage. Consider using `BufferedReader` for better performance with large resources.

```diff
- context.resources.openRawResource(R.raw.collapse_toolbar).readBytes().decodeToString()
+ context.resources.openRawResource(R.raw.collapse_toolbar).bufferedReader().use { it.readText() }
```

Explanation
* readBytes().decodeToString(): This approach reads the entire file into memory as a byte array, which can lead to high memory consumption for large files.
* bufferedReader().use { it.readText() }: This method reads the file line by line (or in chunks) using a buffer, which is more efficient and reduces memory overhead.